rpcserver.go

```diff
@@ -2447,6 +2447,67 @@ func (s *rpcServer) ExpectSidecarChannel(ctx context.Context,
 	return &poolrpc.ExpectSidecarChannelResponse{}, nil
 }
 
+// DecodeSidecarTicket decodes the base58 encoded sidecar ticket into its
+// individual data fields for a more human-readable representation.
+func (s *rpcServer) DecodeSidecarTicket(ctx context.Context,
+	req *poolrpc.SidecarTicket) (*poolrpc.DecodedSidecarTicket, error) {
+
+	// Parse the ticket from its string encoded representation.
+	t, err := sidecar.DecodeString(req.Ticket)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding ticket: %v", err)
+	}
+
+	serializePubKey := func(key *btcec.PublicKey) []byte {
+		if key == nil {
+			return nil
+		}
+
+		return key.SerializeCompressed()
+	}
+
+	resp := &poolrpc.DecodedSidecarTicket{
+		Id:                        t.ID[:],
+		Version:                   uint32(t.Version),
+		State:                     t.State.String(),
+		OfferCapacity:             uint64(t.Offer.Capacity),
+		OfferPushAmount:           uint64(t.Offer.PushAmt),
+		OfferLeaseDurationBlocks:  t.Offer.LeaseDurationBlocks,
+		OfferSignPubkey:           serializePubKey(t.Offer.SignPubKey),
+		OfferAuto:                 t.Offer.Auto,
+		ExecutionPendingChannelId: nil,
```

Comment:
nit: no need to `nil` `ExecutionPendingChannel`.

Reply:
Yes, fixed.